File: examples/python/hello.py
import sys
import json
import urllib.request

# verify version
assert sys.version_info >= (3, 11), f"need python 3.11+, got {sys.version}"
print(f"python {sys.version_info.major}.{sys.version_info.minor}.{sys.version_info.micro}")

# stdlib json round-trip
data = {"status": "ok", "sandbox": True}
assert json.loads(json.dumps(data)) == data
print("json: ok")

# stdlib http server sanity — just import, don't bind a port
import http.server
print("http.server: ok")

print("all checks passed")

File: examples/ruby/hello.rb
# verify ruby version
raise "need ruby 3+, got #{RUBY_VERSION}" unless RUBY_VERSION >= "3"
puts "ruby #{RUBY_VERSION}"

# stdlib json
require 'json'
data = { status: "ok", sandbox: true }
raise "json round-trip failed" unless JSON.parse(JSON.dump(data))["status"] == "ok"
puts "json: ok"

# stdlib net/http import (don't open a socket)
require 'net/http'
puts "net/http: ok"

puts "all checks passed"
